### PR TITLE
Added Test Cases for the Legends

### PR DESCRIPTION
**Pull request by AnnapareddyHarsha**
## Current Behavior

Test Cases added for legends

## New Behavior

Added test Cases added for legends

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/b74a9610dc778d7626b1c492813bee91633890fd'>b74a9610dc778d7626b1c492813bee91633890fd</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-charting/src/components/Legends/Legends.test.tsx [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    const props = {
+      legends,
+    };
+    const component = mount(<Legends {...props} />);
+    component.setProps({ ...props });
+    expect(renderMock).toHaveBeenCalledTimes(2);
+    renderMock.mockRestore();
+  });
+  it('prop changes', () => {
+    const renderMock = jest.spyOn(LegendsBase.prototype, 'render');
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/Legends/Legends.test.tsx:134]: Could you also test 1) legend selected 2) legend overflow

Comment (AnnapareddyHarsha) [reply packages/react-charting/src/components/Legends/Legends.test.tsx:134]: Added Test cases for legends selected and legends overflow

File: packages/react-charting/src/components/Legends/Legends.test.tsx [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+  it('Should mount legends when not empty', () => {
+    wrapper = mount(<Legends legends={legends} />);
+    const legend = wrapper.getDOMNode().querySelectorAll('[class^="legendContainer"]');
+    expect(legend).toBeDefined();
+  });
+
+  it('Should render overflowProps correctly', () => {
+    wrapper = mount(<Legends legends={legends} {...overflowProps} />);
+    const overflowProp = wrapper.getDOMNode().querySelectorAll('[class^="ms-Layer"]');
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/Legends/Legends.test.tsx:105]: Is the ms-layer query selector sufficient.

File: packages/react-charting/src/components/Legends/Legends.test.tsx [hunk 3]
```diff
@@ -100,10 +160,25 @@ describe('AreaChart - basic props', () => {
...
+  it('Should mount Overflow Button when not empty', () => {
+    wrapper = mount(<Legends legends={legends} {...overflowProps} overflowText={'OverFlow Items'} />);
+    const overflowBtnText = wrapper.getDOMNode().querySelectorAll('[class^="ms-OverflowSet-overflowButton"]');
+    expect(overflowBtnText).toBeDefined();
+  });
+
+  it('Should not mount Overflow when empty', () => {
+    wrapper = mount(<Legends legends={legends} {...overflowProps} overflowText={'OverFlow Items'} />);
+    const overflowBtn = wrapper.getDOMNode().querySelectorAll('[class^="ms-OverflowSet-overflowButton"]');
+    expect(overflowBtn!.length).toBe(0);
```
Comment (AtishayMsft) [inline packages/react-charting/src/components/Legends/Legends.test.tsx:172]: How did overflow became empty for this case.

File: packages/react-charting/src/components/Legends/Legends.test.tsx [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+    };
+    const component = mount(<Legends {...props} />);
+    component.setProps({ ...props });
+    expect(renderMock).toHaveBeenCalledTimes(2);
+    renderMock.mockRestore();
+  });
+  it('prop changes', () => {
+    const renderMock = jest.spyOn(LegendsBase.prototype, 'render');
+    const props = {
+      legends,
```
[the remark this answers is not in the document]
Comment (AnnapareddyHarsha) [reply packages/react-charting/src/components/Legends/Legends.test.tsx:211]: Added more legends

File: packages/react-charting/src/components/Legends/Legends.test.tsx (hunk 2 above)
Comment (AnnapareddyHarsha) [reply packages/react-charting/src/components/Legends/Legends.test.tsx]: updated the test case

**Review by AtishayMsft — APPROVED** [review]

File: packages/react-charting/src/components/Legends/Legends.test.tsx (hunk 3 above)
Comment (AnnapareddyHarsha) [reply packages/react-charting/src/components/Legends/Legends.test.tsx:172]: resolving this as I had an external conversation with Atishay
